src/Compression/CompressedReadBufferFromFile.cpp

```diff
@@ -102,17 +104,30 @@ size_t CompressedReadBufferFromFile::readBig(char * to, size_t n)
         size_t size_decompressed = 0;
         size_t size_compressed_without_checksum = 0;
 
-        size_t new_size_compressed = readCompressedData(size_decompressed, size_compressed_without_checksum, false);
+        size_t new_size_compressed = readCompressedDataBlockHold(size_decompressed, size_compressed_without_checksum);
```

Comment:
It's a bit confusing for the reader to understand from the name `readCompressedDataBlockHold()` (and it's documentation which is absent :-) ) what this method is returning. Maybe add more comments here.

Reply:
Please see earlier comment